### PR TITLE
feat: npm publication

### PR DESCRIPTION
**Pull request by Ingwersaft**
Also removed origin which is not needed for Cognito API interaction and made the region configurable (so tests can use cognito clients outside of eu-central-1).

The repository secrets `REGION` and `NPMJS_ACCESS_KEY` are already created, the gitlab workflows are adapted as well :+1: 

EDIT: Also switched to JDK 11 for the build process (Java 8 is still the target compilation version to keep compatibality) as the npm publish gradle plugin is build using Java 11 as target

### CODE REVIEW

File: auth/src/commonMain/kotlin/com/liftric/auth/Configuration.kt [hunk 1]
```diff
@@ -8,7 +8,15 @@ import io.ktor.http.HeadersBuilder
  * Configuration object for the auth handler
  * Holds all headers needed to make requests to AWS Cognito
  */
-class Configuration(origin: String, region: Region, val clientId: String) {
+class Configuration(region: Region, val clientId: String) {
+
+    /**
+     * For compatibility reasons this constructor is needed, previous Auth versions set the Origin header
+     */
```
Comment (benjohnde) [inline auth/src/commonMain/kotlin/com/liftric/auth/Configuration.kt:15]: You could also add the `@Deprecation` annotation.

**Review by benjohnde — COMMENTED** [review]
LGTM!

File: auth/src/commonMain/kotlin/com/liftric/auth/Configuration.kt (hunk 1 above)
Comment (Ingwersaft) [reply auth/src/commonMain/kotlin/com/liftric/auth/Configuration.kt:15]: Good point!

**Review by gaebel — APPROVED** [review]
Nice, good job :)

**Pull request merged**